### PR TITLE
Create layout component #674

### LINKED ISSUE #674: Create a layout component
**Describe the solution you'd like**
Create a layout component that can be reused for all pages. Include the nav bar, set automatic margins and padding, and add the title of the page so that it is consistent throughout the web app.

**Additional context**
my books

Should look similar to this, with the nav bar on the left and the page title to the right of it

### PR DESCRIPTION
**Pull request by ofvoler**
## Summary of change

Add a layout component that formats pages: includes nav bar, page header, and margins/padding for page content.

There is an optional parameter 'centered', which can be used for pages such as Delete Account (#678). The styling for this can be further developed as we have a need for centered pages—currently the only styling is center-aligning the header and text.

I then refactored all four pages to use the new layout component. I was able to get rid of a lot on inconsistent styling with this!

## Additional context

my books
goal
stats
settings

## Related issue

Closes #674 

## Pull request checklist

Please keep this checklist in & ensure you have done the following:

- [x] Read, understood and adhered to our [contributing document](https://github.com/knjk04/book-project/blob/master/CONTRIBUTING.md).
  - [x] Ensure that you were first assigned to a relevant issue before creating this pull request
  - [x] Ensure code changes pass all tests

- [x] Read, understood and adhered to our [style guide](https://github.com/knjk04/book-project/blob/master/STYLEGUIDE.md). A lot of our code reviews are spent on ensuring compliance with our style guide, so it would save a lot of time if this was adhered to from the outset. 

- [x] Filled in the summary, context (if applicable) and related issue section. Replace the square brackets and its placeholder content with your contents. For an example, see any merged in pull request
  - [x] Included a screenshot(s) if a UI change was involved (it may look different on a reviewer's device)

- [x] Created a branch that has a descriptive name (what your branch is for in a few words and includes the issue number at the end, e.g. `test-reading-goal-123`

- [x] Set this pull request to 'draft' if you are still working on it

- [x] Resolved any merge conflicts

For any of the optional checkboxes (e.g. the screenshots one), still check it if it does not apply.

If in doubt, get in touch with us by creating a [Q&A discussion](https://github.com/Project-Books/book-project/discussions/categories/q-a) on GitHub or ask on the #help channel in our Slack workspace

### CODE REVIEW

**Comment by knjk04** [comment]
@ofvoler Thanks for working on this! I've asked @RivaD2 to take a look. I'll also take a look soon

File: frontend/src/shared/book-display/ShelfCarousel.css [hunk 1]
```diff
@@ -9,20 +9,16 @@
...
-}
-
 .shelf-container {
-    padding: 0px 40px;
+    margin-bottom: 40px;
 }
 
 .shelf-title {
     display: inline-block;
     font-size: 20px;
```
Comment (RivaD2) [inline frontend/src/shared/book-display/ShelfCarousel.css:18]: Looks like we can bump the font-size up a tad, to 24px based of specs in Figma :)

File: frontend/src/shared/components/Layout.tsx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+import React, { Component } from 'react'
+import { NavBar } from '../navigation/NavBar'
+import './Layout.css'
+
+export function Layout(props: LayoutProps) {
+    const format = (props.centered ? "centered" : "");
+
+    return (
+        <div className="layoutContainer">
```
Comment (RivaD2) [inline frontend/src/shared/components/Layout.tsx:26]: It looks like kebab-case was used for all case styles in Goal.tsx/Goal.css, MyBooks.tsx/MyBooks.tsx etc.
Can we change classNames in `Layout.css` and `Layout.tsx` to match the other case styles? e.g.(.layout-container, .page-content, .page-title etc.)

**Review by RivaD2 — COMMENTED** [review]
Great work here, really nice! 
I left two comments regarding font-size for shelf-titles and then using kebab case for case styles (so it matches the other file case styles). Once the changes are made, we are good to go!

File: frontend/src/my-books/MyBooks.tsx [hunk 3]
```diff
@@ -26,15 +26,12 @@ interface IState {
 class MyBooks extends Component<Record<string, unknown>, IState> {
     render() {
         return (
-            <div>
-                <NavBar />
-                <div>
-                    <ShelfCarousel title="Reading" />
```
Comment (RivaD2) [inline frontend/src/my-books/MyBooks.tsx:32]: Looks like we can bump the font-size up a tad for the ShelfCarousel titles to 24px based of specs in Figma :)

File: frontend/src/shared/components/Layout.tsx (hunk 2 above)
Comment (ofvoler) [reply frontend/src/shared/components/Layout.tsx:26]: That's a good point! Do you know if kebab-case is the "right" style for style class names? In our [docs site](https://project-books.github.io/docs/styleguides/typescript-styleguide/) it looks like most names should be camelCase, but style class names aren't mentioned. @knjk04 do you have a preference?

Comment (RivaD2) [reply frontend/src/shared/components/Layout.tsx:26]: I also saw that the docs don't mention a preference for it and asked Karan if we should add that in to our docs :)

**Comment by ofvoler** [comment]
Thank you, Riva! I updated the font size, so now we can just wait to see about the case styling.

File: frontend/src/shared/components/Layout.tsx (hunk 2 above)
Comment (knjk04) [reply frontend/src/shared/components/Layout.tsx:26]: We'll go back in a separate issue and adhere to the new style guide. I wouldn't want us to increase the blast radius of this PR

**Review by knjk04 — APPROVED** [review]
Thanks, this is great!

**Review by RivaD2 — APPROVED** [review]

**Comment by RivaD2** [comment]
@knjk04 It looks like I don't have write access yet (I can't merge PR). Can you merge this one in? :)

**Pull request merged**